apps/champions/test/battle_test.exs

```diff
@@ -679,4 +679,69 @@ defmodule Champions.Test.BattleTest do
                Champions.Battle.Simulator.run_battle([unit], [target_dummy_5], maximum_steps: maximum_steps).result
     end
   end
+
+  test "All", %{target_dummy_character: target_dummy_character} do
+    maximum_steps = 1
+
+    # Create a character with a basic skill that will deal 10 damage to all the enemies
+    basic_skill_params =
+      TestUtils.build_skill(%{
+        name: "DealDamage All Enemies",
+        mechanics: [
+          %{
+            trigger_delay: 0,
+            apply_effects_to:
+              TestUtils.build_apply_effects_to_mechanic(%{
+                effects: [
+                  TestUtils.build_effect(%{
+                    executions: [
+                      %{
+                        type: "DealDamage",
+                        attack_ratio: 1,
+                        energy_recharge: 0
+                      }
+                    ]
+                  })
+                ],
+                targeting_strategy: %{
+                  type: "all",
+                  target_allies: false
+                }
+              })
+          }
+        ],
+        cooldown: maximum_steps * @miliseconds_per_step - 1
+      })
+
+    {:ok, character} =
+      TestUtils.build_character(%{
+        name: "Character",
```

Comment:
Name this character with a more specific name, so as to not get any "name taken" errors. Same with the "Empty Skill". You could name them "All Character" and "All Empty Skill".

Reply:
Also fixed!